### PR TITLE
[feat] support conda platforms

### PR DESCRIPTION
**Pull request by nh13**
Support specifying conda platforms (e.g. from conda-lock):

 For example
```
   - conda:
     channels:
       - conda-forge
       - bioconda
     requirements:
       - samtools
       - fgbio=1.1.0
     platforms:
       - linux-32
       - osx-arm64
       - win32
```

Implements #30.

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -142,6 +143,7 @@ object tools extends CommonModule {
     ivy"io.circe::circe-yaml::$circeYamlVersion",
     ivy"io.circe::circe-generic::$circeGenericVersion",
     ivy"io.circe::circe-generic-extras::$circeGenericExtrasVersion",
+    ivy"com.beachape::enumeratum::$enumeratumVersion",
```
Comment (nh13) [inline build.sc]:
```suggestion
```

File: build.sc [hunk 2]
```diff
@@ -21,6 +21,7 @@ private val fgSoptVersion             = "1.1.0-e565813-SNAPSHOT"
 private val circeYamlVersion          = "0.13.0"
 private val circeGenericVersion       = "0.13.0"
 private val circeGenericExtrasVersion = "0.13.0"
+private val enumeratumVersion         = "1.7.2"
```
Comment (nh13) [inline build.sc]:
```suggestion
```

**Review by leeleavitt — APPROVED** [review]
This is exactly what we need to enable the conda-lock locking feature. Thank you for adding this so fast

**Comment by leeleavitt** [comment]
I'm going to test it locally now.

**Comment by leeleavitt** [comment]
I built off this branch and compile the jar locally after installing `brew install mill`

From there I built it via,

`mill tools.localJar`

I then copied the `./jars/jar/conda-env-builder.jar` from the repo into the conda environment where i needed this update, `.dev/share/conda-env-builder/conda-env-builder.jar`

## My input:

`test.yaml`
```
name: a_pipeline
environments:
  defaults:
    steps:
      - conda:
          platforms:
            - linux-64
            - osx-64
          channels:
            - bioconda
            - conda-forge
            - bioconda/label/broken
            - conda-forge/label/broken
          requirements:
            - python=3.6.4
            - pip=21.3.1
            - pysam=0.15.3
  common:
    steps:
      - conda:
          requirements:
            - python
            - pip
            - pysam
  validation:
    inherits:
      - common
 ```
 
 ## My command
 `conda-env-builder --mamba Assemble --config test.yaml --output test/ --overwrite -n validation`
 
 and my output is the same,
 
 ## my output
`test/validation.yml`
 ```
 name: validation
channels:
  - bioconda
  - conda-forge
  - bioconda/label/broken
  - conda-forge/label/broken
dependencies:
  - python=3.6.4
  - pip=21.3.1
  - pysam=0.15.3
```

**Comment by leeleavitt** [comment]
Hi Nils, it seems that I need to make an additional change for the previous pull request to work. My apologies for approving it before completing a thorough test on my end. I forked the repo, made the necessary changes, and tested them, and this last change was the final update required.


This needs to be added for the platforms to be added to the output

https://github.com/conda-incubator/conda-env-builder/commit/82fa1af9a7e4a0108ef82ba7e29de6433238943b

**Pull request merged**